plugins/producer_plugin/test/test_trx_full.cpp

```diff
@@ -0,0 +1,212 @@
+BOOST_AUTO_TEST_CASE(producer) {
+   boost::filesystem::path temp = boost::filesystem::temp_directory_path() / boost::filesystem::unique_path();
+
+   try {
+      std::promise<std::tuple<producer_plugin*, chain_plugin*>> plugin_promise;
+      std::future<std::tuple<producer_plugin*, chain_plugin*>> plugin_fut = plugin_promise.get_future();
+      std::thread app_thread( [&]() {
+         fc::logger::get(DEFAULT_LOGGER).set_log_level(fc::log_level::debug);
+         std::vector<const char*> argv =
+               {"test", "--data-dir", temp.c_str(), "--config-dir", temp.c_str(),
+                "-p", "eosio", "-e", "--max-transaction-time", "475", "--disable-subjective-billing=true" };
+         appbase::app().initialize<chain_plugin, producer_plugin>( argv.size(), (char**) &argv[0] );
+         appbase::app().startup();
+         plugin_promise.set_value(
+               {appbase::app().find_plugin<producer_plugin>(), appbase::app().find_plugin<chain_plugin>()} );
+         appbase::app().exec();
+      } );
+
+      auto[prod_plug, chain_plug] = plugin_fut.get();
+      auto chain_id = chain_plug->get_chain_id();
+
+      std::deque<block_state_ptr> all_blocks;
+      std::promise<void> empty_blocks_promise;
+      std::future<void> empty_blocks_fut = empty_blocks_promise.get_future();
+      std::atomic<size_t> num_aborts = 0;
+      auto ab = chain_plug->chain().accepted_block.connect( [&](const block_state_ptr& bsp) {
+         static int num_empty = std::numeric_limits<int>::max();
+         all_blocks.push_back( bsp );
+         if( bsp->block->transactions.empty() ) {
+            --num_empty;
+            if( num_empty == 0 ) empty_blocks_promise.set_value();
+         } else { // we want a few empty blocks after we have some non-empty blocks
+            num_empty = 10;
+         }
+      } );
+      auto ba = chain_plug->chain().block_abort.connect( [&]( uint32_t bn ) {
+         ++num_aborts;
+      } );
+      auto bs = chain_plug->chain().block_start.connect( [&]( uint32_t bn ) {
+      } );
+
+      std::deque<packed_transaction_ptr> trxs;
+      std::atomic<size_t> next_calls = 0;
+      std::atomic<size_t> num_posts = 0;
+      std::atomic<size_t> trace_with_except = 0;
+      std::atomic<bool> trx_match = true;
+      const size_t num_pushes = 4242;
+      for( size_t i = 1; i <= num_pushes; ++i ) {
+         auto ptrx = make_unique_trx( chain_id );
+         dlog( "posting ${id}", ("id", ptrx->id()) );
+         app().post( priority::low, [ptrx, &next_calls, &num_posts, &trace_with_except, &trx_match, &trxs]() {
+            ++num_posts;
+            bool return_failure_traces = false; // not supported in version 2.1.x, in 2.2.x+ = num_posts % 2 == 0;
+            app().get_method<plugin_interface::incoming::methods::transaction_async>()(ptrx,
+               false, // persist_until_expiried
+               // 2.2.x+ false, // read_only
+               // 2.2.x+ return_failure_traces == true, // return_failure_traces
```

Comment:
For posterity: `develop-boxed` has a unique implementation of return_failure_traces, so it's not in this method.